tests/unit/verifiers/test_ovn_central.py

```diff
@@ -0,0 +1,783 @@
+    """Verify expected result from consistency check."""
+    # Mock complete_cluster_status property
+    complete_cluster_status = {
+        unit: ovn_central.UnitClusterStatus(**status)
+        for unit, status in sample_cluster_data.items()
+    }
+    cluster_status_property = PropertyMock(return_value=complete_cluster_status)
+
+    mocker.patch.object(
+        ovn_central.OvnCentral,
+        "complete_cluster_status",
+        new_callable=cluster_status_property,
+    )
+
+    # create verifier and run check
+    verifier = ovn_central.OvnCentral([Unit("ovn-central/0", model)])
+
+    result = verifier.check_leader_consistency()
+    assert result == expected_result
+
+
+#  Tests for ovn_central.ClusterStatus class
+def test_cluster_status_init(ovn_cluster_status_raw, ovn_cluster_status_dict):
+    """Test successful initialization of ClusterStatus class from data string."""
+    status = ovn_central.ClusterStatus(ovn_cluster_status_raw)
+    expected_status = ovn_cluster_status_dict
+
+    assert status.cluster_id == expected_status["Cluster ID"]
+    assert status.server_id == expected_status["Server ID"]
+    assert status.status == expected_status["Status"]
+    assert status.role == expected_status["Role"]
+    assert status.vote == expected_status["Vote"]
+    assert status.log == expected_status["Log"]
+    assert status.entries_not_committed == expected_status["Entries not yet committed"]
+    assert status.entries_not_applied == expected_status["Entries not yet applied"]
+    assert status.servers == expected_status["Servers"]
+
+
+def test_cluster_status_init_missing_key(ovn_cluster_status_dict):
+    """Test failure of ClusterStatus initialization when data string is missing key."""
+    missing_key = "Status"
+    ovn_cluster_status_dict.pop(missing_key)
+    status_string = yaml.dump(ovn_cluster_status_dict)
+    with pytest.raises(ovn_central.JujuVerifyError) as exc_info:
+        _ = ovn_central.ClusterStatus(status_string)
+
+    assert (
+        str(exc_info.value) == f"Failed to deserialize OVN cluster status. Missing "
+        f"'{missing_key}' from cluster status data."
+    )
+
+
+def test_cluster_status_init_yaml_fail():
+    """Test failure of ClusterStatus initialization when data string is not yaml."""
+    data = "not a\nyaml"
+    with pytest.raises(ovn_central.JujuVerifyError) as exc_info:
+        _ = ovn_central.ClusterStatus(data)
+
+    assert str(exc_info.value) == f"Failed to load OVN status as YAML:\n{data}"
+
+
+def test_cluster_status_short_id(ovn_cluster_status):
+    """Test that short_id property returns first 4 chars of server_id."""
+    expected_short_id = ovn_cluster_status.server_id[:4]
+
+    assert ovn_cluster_status.short_id == expected_short_id
+
+
+@pytest.mark.parametrize(
+    "leader_identity, is_leader",
+    [
+        ("self", True),  # Unit is cluster leader if it reports leader as "self"
+        ("ac5b", False),  # Unit is not cluster leader if it reports another unit's ID
```

Comment:
I will add some reasonable cornor cases. ex:  "None" and ""(empty) string.

Reply:
This got me thinking that maybe there should be a validation when ClusterStatus instance is created to make sure that all attributes have expected values and not None/empty strings.